sdk/trace/src/main/java/io/opentelemetry/sdk/trace/AndroidFriendlyRandomIdGenerator.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace;
+
+import io.opentelemetry.api.trace.SpanId;
+import io.opentelemetry.api.trace.TraceId;
+import java.util.Random;
+
+/**
+ * {@link IdGenerator} instance that doesn't use {@link java.util.concurrent.ThreadLocalRandom},
+ * which is broken on most versions of Android (it uses the same seed everytime it starts up).
+ */
+enum AndroidFriendlyRandomIdGenerator implements IdGenerator {
+  INSTANCE;
+
+  // note: we can't use ThreadLocal.withInitial(..) on Android API 21
+  private static final ThreadLocal<Random> threadLocalRandom =
```

Comment:
Maybe better to avoid the thread local and just go with a static final, I don't think we'd expect such contention for it to win over normal access on Android

Reply:
Hm. Interesting thought. You're probably right in normal use-cases. I wish there were a good way to measure that outside an actual Android VM.

Reply:
I'll at least test this out locally on an Android app just to see if there is anything obviously bad about just using a static final.

Reply:
seems to work fine in my limited testing; I'll change to the simpler implementation. Thanks for the suggestion!